modules/wallet/encrypt.go

```diff
@@ -193,29 +193,48 @@ func (w *Wallet) managedUnlock(masterKey crypto.TwofishKey) error {
 	subscribed := w.subscribed
 	w.mu.RUnlock()
 	if !subscribed {
-		// Subscription can take a while, so spawn a goroutine to print the
-		// wallet height every few seconds. (If subscription completes
-		// quickly, nothing will be printed.)
-		done := make(chan struct{})
-		go w.rescanMessage(done)
-		defer close(done)
-
-		err = w.cs.ConsensusSetSubscribe(w, lastChange)
-		if err == modules.ErrInvalidConsensusChangeID {
-			// something went wrong; resubscribe from the beginning
-			err = dbPutConsensusChangeID(w.dbTx, modules.ConsensusChangeBeginning)
+		if primarySeedProgress == 0 {
+			// If the wallet has not been used, we don't need to scan the entire
+			// blockchain looking for relevant transactions. Instead, just grab
+			// the current height and subscribe to future blocks.
+			err = w.cs.ConsensusSetSubscribe(w, modules.ConsensusChangeRecent)
 			if err != nil {
-				return fmt.Errorf("failed to reset db during rescan: %v", err)
+				return fmt.Errorf("wallet subscription failed: %v", err)
 			}
-			err = dbPutConsensusHeight(w.dbTx, 0)
+			w.mu.Lock()
+			err = dbPutConsensusHeight(w.dbTx, w.cs.Height())
+			w.mu.Unlock()
```

Comment:
you shouldn't be calling `cs.Height()` if you are trying to stay synchronized with the most recent consensus change.

The cc doesn't return the value of the inputs? I would be worried about breaking compatibility with other apps that have already been written. For example, by making this change we may break our exchanges.

We can't suddenly start serving values of '0' out of the api where there used to be non-zero values. We have a lot of people (including several exchanges) using the wallet api and we don't know which calls they are depending on, but we do know that often it's more of the calls than we thought necessary.